enhancements/security/x509-trust.md

```diff
@@ -0,0 +1,213 @@
+
+### Per-component Trust
+
+Each component will define its own well-known ConfigMap(s) in its own namespace, with the same `ca-bundle.crt` and `composition` keys as [the cluster-scoped ConfigMap](#cluster-scoped-default-trust).
+For example, the registry operator uses [`trusted-ca`][registry-configmap-name] in [the `openshift-image-registry` namespace][registry-configmap-namespace].
+Components that have distinct trust domains may define multiple ConfigMaps for each domain.
+For example, there may be one ConfigMap for proxy/egress TLS, and another ConfigMap for [S/MIME][smime] verification.
+To populate that trust bundle, the cluster administrator can set labels on the ConfigMap(s).
+For example, [the `config.openshift.io/inject-proxy-cabundle` label](../proxy/global-cluster-egress-proxy.md#implementation-detailsnotesconstraints-optional) asks for the current proxy trust bundle.
+
+The cluster-version operator (CVO) [merges ConfigMaps][cluster-version-operator-EnsureConfigMap] by [clobbering any manifest-defined data][cluster-version-operator-mergeMap], [labels, and annotations][cluster-version-operator-EnsureObjectMeta].
+Data keys, labels, and annotations not defined in the manifest are ignored.
+This means that components which decide to provide their per-component ConfigMaps via the CVO should not set labels like `config.openshift.io/inject-default-cabundle` unless they also set [the `release.openshift.io/create-only` annotation to `true`][cluster-version-operator-create-only], to allow cluster administrators to manipulate the per-component ConfigMap without the CVO stomping on their updates.
+For example, the registry operator [sets `create-only`][registry-configmap-create-only].
+
+#### Universal Base Image Containers
+
+FIXME: UBI-specific docs for this [RHEL behavior](rhel-cert-injection) that show that you don't need to install some non-UBI RHEL stuff to make it work.
+
+Components can mount `ca-bundle.crt` from their ConfigMap(s) under `/usr/share/pki/ca-trust-source/anchors/`.
+Components are then responsible for running `update-ca-trust` (FIXME: do we need `extract` or not?) at start.
+They must also run `update-ca-trust` periodically thereafter to pull in any [updates from the mounted ConfigMap][mounted-configmap-updates].
+
+Components that need a single trust bundle injected can bypass `update-ca-trust` and mount their `ca-bundle.crt` directly into `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem` (FIXME: RHEL/UBI docs confirming this as a supported workflow).
+For example, the registry operator [does this][registry-configmap-mount].
+
+FIXME: We need to grow a way to handle `override` `composition`.
+
+### Node Trust
+
+[Red Hat Enterprise Linux CoreOS (RHCOS)][rhcos], and nodes more broadly, have a separate trust-management framework getting hashed out [here][machine-config-operator-trust].
+
+### Container Image Registry Trust
+
+FIXME: fill in how this works
+
+### User Stories
```

Comment:
> Would love to see a user story for a cloud platform (e.g OpenStack) using self-signed certs.

Sounds good.  Anyone familiar enough to be able to float some wording?  @iamemilio?

Reply:
> As an OpenStack engineer, I want to trust the additionalCertBundle in containers running services that have to authenticate against OpenStack so that clusters with self signed certificates can succesfully install OpenShift.

How is this?

Reply:
> As an OpenStack engineer...

I'd rather hang the use-case on the cluster admistrator we're enabling.  Took a stab at this with 47290b8 -> 4a3930d, comment on the new story if you have further wordsmithing suggestions :).